qt/python/mantidqt/project/recovery/projectrecovery.py

```diff
@@ -0,0 +1,471 @@
+                               "SaveGEMMAUDParamFile", "SaveGSASInstrumentFile", "SaveGSS", "SaveHKL",
+                               "SaveILLCosmosAscii", "SaveISISNexus", "SaveIsawDetCal", "SaveIsawPeaks",
+                               "SaveIsawQvector", "SaveIsawUB", "SaveLauenorm", "SaveMD", "SaveMDWorkspaceToVTK",
+                               "SaveMask", "SaveNISTDAT", "SaveNXSPE", "SaveNXTomo", "SaveNXcanSAS", "SaveNexus",
+                               "SaveNexusPD", "SaveNexusProcessed", "SaveOpenGenieAscii", "SavePAR", "SavePDFGui",
+                               "SavePHX", "SaveParameterFile", "SavePlot1D", "SavePlot1DAsJson", "SaveRKH",
+                               "SaveReflCustomAscii", "SaveReflThreeColumnAscii", "SaveReflections",
+                               "SaveReflectometryAscii", "SaveSESANS", "SaveSPE", "SaveTBL", "SaveToSNSHistogramNexus",
+                               "SaveVTK", "SaveVulcanGSS", "SaveYDA", "SaveZODS"]
+        # If you want to ignore an algorithms' property, then add to the string below in the format:
+        # "AlgorithmName + PropertyName". The final string should look like "a + b , c + d, ...".
+        # This uses string representation to pass to the C++ algorithm. The outer delimiter is `,`
+        # and the inner delimiter is `+` for the list of lists. e.g. [[a, b],[c, d]] = "a + b , c + d".
+        self.alg_properties_to_ignore = "StartLiveData + MonitorLiveData"
+
+        # The recovery GUI's presenter is set when needed
+        self.recovery_presenter = None
+
+        self.thread_on = False
+
+        # Set to true by workbench on close to kill the thread on completion of project save
+        self.closing_workbench = False
+
+    ######################################################
+    #  Utility
+    ######################################################
+
+    def start_recovery_thread(self):
+        if not self.recovery_enabled:
+            logger.debug("Project Recovery: Recovery thread not started as recovery is disabled")
+            return
+
+        if not self.thread_on:
+            self._timer_thread.start()
+            self.thread_on = True
+
+    def stop_recovery_thread(self):
+        if self._timer_thread is not None:
+            self._timer_thread.cancel()
+            self.thread_on = False
+
+    @staticmethod
+    def _remove_empty_folders_from_dir(directory):
+        folders = glob(os.path.join(directory, "*", ""))
+        for folder in folders:
+            try:
+                os.rmdir(folder)
+            except OSError:
+                # Fail silently as expected for all folders
+                pass
+
+    def _remove_directory_and_directory_trees(self, directory):
+        if directory is None or not os.path.exists(directory):
+            return
+
+        # Only allow deleting in subdirectories of workbench-recovery
+        if self.recovery_directory not in directory:
```

Comment:
Good idea to check this. Could be slightly stricter and check that directory is strictly a subdirectory of the recovery folder?

Reply:
The utility function is used in many places that could ask to delete the hostname directory which is inside this directory call. If the hostname folder is empty then it will attempt to delete it and potentially (I don't remember the exact implementation, the `workbench-recovery` directory)